src/vivarium_public_health/risks/effect.py

```diff
@@ -316,3 +321,185 @@ def _get_risk_exposure_class(self, builder: Builder) -> Risk:
                 f"Risk effect model {self.name} requires a Risk component named {self.risk}"
             )
         return risk_exposure_component
+
+
+class NonLogLinearRiskEffect(RiskEffect):
+    """A component to model the impact of an exposure-parametrized risk factor on
+    the target rate of some affected entity. This component will
+
+    1) read TMRED data from the artifact and define the TMREL
+    2) calculate the relative risk at TMREL by linearly interpolating over
+       relative risk data defined in the configuration
+    3) divide relative risk data from configuration by RR at TMREL
+       and clip to be greater than 1
+    4) build a LookupTable which returns the exposure and RR of the left and right edges
+       of the RR bin containing a simulant's exposure
+    5) use this LookupTable to modify the target pipeline by linearly interpolating
+       a simulant's RR value and multiplying it by the intended target rate
+    """
+
+    ##############
+    # Properties #
+    ##############
+
+    @property
+    def configuration_defaults(self) -> Dict[str, Any]:
+        """
+        A dictionary containing the defaults for any configurations managed by
+        this component.
+        """
+        return {
+            self.name: {
+                "data_sources": {
+                    "relative_risk": f"{self.risk}.relative_risk",
+                    "population_attributable_fraction": f"{self.risk}.population_attributable_fraction",
+                },
+            }
+        }
+
+    @property
+    def columns_required(self) -> list[str]:
+        return [f"{self.risk.name}_exposure"]
+
+    #################
+    # Setup methods #
+    #################
+
+    def build_all_lookup_tables(self, builder: Builder) -> None:
+        rr_data = self.get_relative_risk_data(builder)
+
+        # check that rr_data has 1000 parameter values
+        exposure_values = rr_data["parameter"].values
+        if len(np.unique(exposure_values)) != NUM_RR_EXPOSURE_VALUES:
```

Comment:
Is this constant being used other places? If it IS NOT then maybe this could be configurable.

Reply:
Honestly thinking about it now this seems unnecessarily restrictive. I was assuming all relative risk data for this component would come from GBD but I think its _possible_ that a researcher would want to define their own custom data. I see no harm in skipping this test and replacing it with checking for numeric data in the parameter column. That's also a better test because the component won't break if we have greater or fewer exposure values but if will if the data is not numeric.